extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/reactive/ReactiveMongoCollection.java

```diff
@@ -0,0 +1,1410 @@
+package io.quarkus.mongodb.reactive;
+
+import java.util.List;
+
+import org.bson.Document;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
+
+import com.mongodb.MongoNamespace;
+import com.mongodb.bulk.BulkWriteResult;
+import com.mongodb.client.model.BulkWriteOptions;
+import com.mongodb.client.model.CountOptions;
+import com.mongodb.client.model.CreateIndexOptions;
+import com.mongodb.client.model.DeleteOptions;
+import com.mongodb.client.model.DropIndexOptions;
+import com.mongodb.client.model.EstimatedDocumentCountOptions;
+import com.mongodb.client.model.FindOneAndDeleteOptions;
+import com.mongodb.client.model.FindOneAndReplaceOptions;
+import com.mongodb.client.model.FindOneAndUpdateOptions;
+import com.mongodb.client.model.IndexModel;
+import com.mongodb.client.model.IndexOptions;
+import com.mongodb.client.model.InsertManyOptions;
+import com.mongodb.client.model.InsertOneOptions;
+import com.mongodb.client.model.RenameCollectionOptions;
+import com.mongodb.client.model.ReplaceOptions;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.model.WriteModel;
+import com.mongodb.client.model.changestream.ChangeStreamDocument;
+import com.mongodb.client.result.DeleteResult;
+import com.mongodb.client.result.UpdateResult;
+import com.mongodb.reactivestreams.client.ClientSession;
+
+import io.quarkus.mongodb.AggregateOptions;
+import io.quarkus.mongodb.ChangeStreamOptions;
+import io.quarkus.mongodb.DistinctOptions;
+import io.quarkus.mongodb.FindOptions;
+import io.quarkus.mongodb.MapReduceOptions;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+/**
+ * A reactive API to interact with a Mongo collection.
+ *
+ * @param <T> The type that this collection will encode documents from and decode documents to.
+ * @since 1.0
+ */
+public interface ReactiveMongoCollection<T> {
+
+    /**
+     * Gets the namespace of this collection.
+     *
+     * @return the namespace
+     */
+    MongoNamespace getNamespace();
+
+    /**
+     * Get the class of documents stored in this collection.
+     *
+     * @return the class
+     */
+    Class<T> getDocumentClass();
+
+    /**
+     * Gets an estimate of the count of documents in a collection using collection metadata.
+     *
+     * @return n {@link Uni} completed with the estimated number of documents
```

Comment:
Missing an `a`  in the javadoc ?
`@return an {@link Uni} completed with the estimated number of documents`

Reply:
"replace all" is a broken pattern! Fixing...

Reply:
Fixed.